### ISSUE TITLE
[mat-slider] Screen reader users do not get notified when step is too small

### ISSUE THREAD

**Issue by mertdeg2** [angular/components#14328]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Screen reader announces the changes when user presses left/right arrows to change the slider value

#### What is the current behavior?
For min=0,max=1, and step 0.1 = everything works as expected
For min=0,max=1, and step 0.01 = screen reader doesn't announce the changed value.

Tested this with VoiceOver and JAWS

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-muqy2t-hwoxtw?file=app/slider-overview-example.html

Video showing step size 0.1 with voiceover= 
![unnamed screen capture 1](https://user-images.githubusercontent.com/16231118/49253763-40c7f100-f3dc-11e8-91fd-108c56b478dc.gif)
Video showing step size 0.01 with voiceover =
![unnamed screen capture 3](https://user-images.githubusercontent.com/16231118/49253961-ce0b4580-f3dc-11e8-9f57-a88e9daf7a90.gif)

#### What is the use-case or motivation for changing an existing behavior?
Better a11y

**Comment by mertdeg2** [comment]
@jelbourn

**Comment by jelbourn** [comment]
@mertdeg2 does this work with *any* slider? I suspect this would be a screen-reader limitation and not something specific to `mat-slider`.

**Comment by jelbourn** [comment]
@mertdeg2 I created a simplified slider with no Angular involved:
https://stackblitz.com/edit/typescript-cmsveq?file=index.ts

I just tested with VoiceOver, but it does seem that the screen-reader itself does not handle those small values. However, as a workaround, you can also add `aria-valuetext` which *is* read. I've added this as an item for my next sync with the Google Accessibility team since I want to get their official recommendation. In the meantime, you could apply `aria-valuetext` directly as a workaround, just be sure to correctly format the result since the floating point arithmetic may otherwise  lead to reading out something like "0.05000000000000000001"

**Comment by mmalerba** [comment]
Closing as this seems unrelated to Angular
